front/src/applications/operationalStudies/hooks/useManageTrainScheduleContext.tsx

```diff
@@ -29,11 +34,27 @@ export const ManageTrainScheduleContextProvider = ({
 
   const { launchPathfinding, pathfindingState, infraInfo } = usePathfinding(setPathProperties);
 
+  const { getPathSteps } = useOsrdConfSelectors();
+  const pathSteps = useSelector(getPathSteps);
+
   const voltageRanges = useMemo(
     () => getPathVoltages(pathProperties?.electrifications, pathProperties?.length),
     [pathProperties]
   );
 
+  useEffect(() => {
+    if (pathProperties) {
+      const allWaypoints = upsertPathStepsInOPs(
+        pathProperties.suggestedOperationalPoints,
+        compact(pathSteps)
+      );
+      setPathProperties({
+        ...pathProperties,
+        allWaypoints,
+      });
+    }
+  }, [pathSteps]);
```

Comment:
Why is this necessary now? What does this fix?

Reply:
AllWaypoints is used to compute the inputs table.
Without this useEffect, if I add some vias from the power restrictions selector, then they are not added in allWaypoints and not displayed in the inputs table.

Note: I think it would be better to have a useMemo for allWaypoints, rather than to store them artificially in pathProperties

Reply:
Agreed! 

(Side note, this `useEffect()` depends on `pathProperties.suggestedOperationalPoints`, but that's missing from the dependencies.)

Reply:
> Note: I think it would be better to have a useMemo for allWaypoints, rather than to store them artificially in pathProperties

Done in https://github.com/OpenRailAssociation/osrd/pull/10169